src/datasets/makeRegression.ts

```diff
@@ -0,0 +1,83 @@
+import { Scikit1D, Scikit2D } from '../types'
+import { min } from 'mathjs'
+import { tf } from '../shared/globals'
+import { makeLowRankMatrix } from './makeLowRankMatrix'
+
+interface MakeRegressionInput {
+  nSamples?: number
+  nFeatures?: number
+  nInformative?: number
+  nTargets?: number
+  bias?: number
+  effectiveRank?: number | null
+  tailStrength?: number
+  noise?: number
+  shuffle?: boolean
+  coef?: boolean
+}
+
+type MakeRegressionOutput =
+  | [Scikit2D, Scikit1D]
+  | [Scikit2D, Scikit1D, Scikit1D]
```

Comment:
`Scikit1D` and `Scikit2D` are meant as input types which are as wide as possible. But as return types, `Tensor1D` and `Tensor2D` might be more practical (allows You to call methods like `X.print()` on them).

Also if `nTargets != 1`, then the return type would be `[Tensor2D, Tensor2D]` and `[Tensor2D, Tensor2D, Tensor2D]` respectively.